test/new_relic/newrelicyml_test.rb

```diff
@@ -0,0 +1,144 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require_relative '../test_helper'
+require_relative '../../lib/tasks/helpers/newrelicyml'
+
+class NewRelicYMLTest < Minitest::Test
+  def test_public_config_true
+    assert NewRelicYML.public_config?(FAKE_CONFIGS[:begonia])
+  end
+
+  def test_public_config_false
+    refute NewRelicYML.public_config?(FAKE_CONFIGS[:pothos])
+  end
+
+  def test_default_value_use_documentation_default
+    assert_equal 2, NewRelicYML.default_value(:lily, FAKE_CONFIGS[:lily])
+  end
+
+  def test_default_value_string
+    assert_equal '""', NewRelicYML.default_value(:monstera, FAKE_CONFIGS[:monstera])
+  end
+
+  def test_default_value_nil
+    assert_equal 'nil', NewRelicYML.default_value(:begonia, FAKE_CONFIGS[:begonia])
```

Comment:
Is the 'nil' string expected? I would have thought either `''` or `nil` would be here.

Reply:
Yes! Unless I use the string version of `nil` or `""`, a blank shows up in `newrelic.yml` since its literally using nil.